### PR TITLE
Allow passing starting centroids to RiemannianKMeans

### LINKED ISSUE #1500: Allow passing initial centroids to RiemannianKMeans via init argument
The class `RiemannianKMeans` has an `init` argument but its value appears to be ignored and the only documented choice is `"random"`. It would be very helpful if initial centroids could be set via this argument. That is, the argument should accept an array of shape `(self.n_clusters, X.shape[1])` that is assigned to `self.centroids` at the start of `fit`.

### PR DESCRIPTION
**Pull request by Viech**
## Checklist

- [x] My pull request has a clear and explanatory title.
- [x] If neccessary, my code is [vectorized](https://www.geeksforgeeks.org/vectorization-in-python/).
- [x] I have added apropriate unit tests.
- [x] I have made sure the code passes all unit tests. (refer to comment below)
- [x] My PR follows [PEP8](https://peps.python.org/pep-0008/) guidelines. (refer to comment below)
- [x] My PR follows [geomstats coding style](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#coding-style-guidelines) and API.
- [ ] My code is properly documented and I made sure the documentation renders properly. ([Link](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#documentation))
- [x] I have linked to issues and PRs that are relevant to this PR.

## Description 

Sometimes users have domain-specific knowledge on where to expect cluster centers. This MR lets them provide initial cluster centers to nudge k-means towards this guess. If the guess is good, this improves the convergence rate and produces a clustering matching the user's expectation.

**I updated the documentation but failed to compile it due to many unrelated errors.** I also had to run the relevant unit tests manually due to issues with the current version of the `werkzeug` library.

## Issue

This closes #1500.

### CODE REVIEW

**Comment by Viech** [comment]
The test failures seem to be for `tests/tests_geomstats/test_discrete_curves.py`, which should be unrelated to this MR.

File: geomstats/learning/kmeans.py [hunk 1]
```diff
@@ -103,24 +105,43 @@ def fit(self, X):
...
+                    "Dimensions of initial centroids and training data do not match."
+                )
+
             centroids = [
-                gs.expand_dims(X[randint(0, n_samples - 1)], 0)
-                for i in range(self.n_clusters)
+                gs.expand_dims(self.init[i], 0) for i in range(self.n_clusters)
             ]
+
         self.centroids = gs.concatenate(centroids, axis=0)
```
Comment (lpereira95) [inline geomstats/learning/kmeans.py:145]: Here I think we can do:

```
self.centroids = gs.copy(centroids)
self.init_centroids = gs.copy(centroids)
```

And above, within the `if` branch that handles `str`, do: `centroids = gs.concatenate(centroids, axis=0)`.

So we simplify the new if branch by simply doing: `centroids = init` (no need for `gs.expand_dims`).

File: tests/tests_geomstats/test_riemannian_kmeans.py [hunk 2]
```diff
@@ -83,3 +83,23 @@ def test_hypersphere_kmeans_initialization(self):
         result = centroids.shape
         expected = (n_clusters, 3)
         self.assertAllClose(expected, result)
+
+    def test_hypersphere_kmeans_init_array(self):
+        gs.random.seed(55)
+
+        n_features = 3
+        n_clusters = 3
```
Comment (lpereira95) [inline tests/tests_geomstats/test_riemannian_kmeans.py:91]: Use a different number of clusters and features, otherwise the shape is `3x3`, which tells very little about the correctness of the result.

**Review by lpereira95 — CHANGES_REQUESTED** [review]
Thanks for the nice contribution @Viech! I have small comments I would like you to address, but they're very small details.

@ninamiolane, this contributions helps to make this method closer to [`sklearn`'s](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html?highlight=kmeans#sklearn.cluster.KMeans). In terms of `init`, we are missing the option `callable`, but it should be trivial to add.

I have some follow-ups for this case, specially regarding testing. (I would also like to create a `_init_centroids` method - following sklearn -, that will simplify testing) If you agree with, I'll implement it after we merge this.

File: geomstats/learning/kmeans.py (hunk 1 above)
Comment (Viech) [reply geomstats/learning/kmeans.py:145]: Makes sense. I also implemented the `callable` option. Note that my approach requires Python >= 3.2 as I'm using the `callable()` built-in function.

File: tests/tests_geomstats/test_riemannian_kmeans.py (hunk 2 above)
Comment (Viech) [reply tests/tests_geomstats/test_riemannian_kmeans.py:91]: Sure. I also added a helper function to make it less redundant.

File: geomstats/learning/kmeans.py (hunk 1 above)
Comment (lpereira95) [reply geomstats/learning/kmeans.py]: Thanks! We assume Python >= 3.7, so it is perfectly fine.

File: geomstats/learning/kmeans.py [hunk 3]
```diff
@@ -103,26 +108,50 @@ def fit(self, X):
...
+                raise ValueError("Need as many initial centroids as clusters.")
+
+            if centroids.shape[1] != X.shape[1]:
+                raise ValueError(
+                    "Dimensions of initial centroids and training data do not match."
+                )
+
+            centroids = gs.copy(centroids)
+
+        self.centroids = centroids
```
Comment (lpereira95) [inline geomstats/learning/kmeans.py:153]: Here we need to do:

```
self.centroids = gs.copy(centroids)
self.init_centroids = gs.copy(centroids)
```

otherwise both `self.centroids` and `self.init_centroids` will be the same object (at least in numpy the arrays are mutable).

File: tests/tests_geomstats/test_riemannian_kmeans.py [hunk 4]
```diff
@@ -66,20 +66,46 @@ def test_hypersphere_kmeans_predict(self):
         )
         self.assertAllClose(expected, result)
 
-    def test_hypersphere_kmeans_initialization(self):
-        gs.random.seed(55)
+    def _test_hypersphere_kmeans_init(
```
Comment (lpereira95) [inline tests/tests_geomstats/test_riemannian_kmeans.py:69]: Nice!

**Review by lpereira95 — CHANGES_REQUESTED** [review]
After changing the little detail about mutability of `self.centroids` and `self.init_centroids`, it is ready to go from my side.

Thanks for the last additions, they really complete this part of the code.

**Review by lpereira95 — APPROVED** [review]
